### PR TITLE
Add up to date index mgmnt examples for dev-reorg

### PR DESCRIPTION
**Pull request by maria-robobug**
Docs issue: [DOC-9209](https://issues.couchbase.com/browse/DOC-9209)

### CODE REVIEW

File: modules/hello-world/examples/IndexHelloWorld.csx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		var cluster = await Cluster.ConnectAsync("couchbase://localhost", "Administrator", "password");
+
+		{
+			Console.WriteLine("[primary]");
+			// tag::primary[]
+			try
+			{
+				// NOTE: 
+				// * Bucket names containing a `-` need to be escaped, see https://issues.couchbase.com/browse/NCBC-2955.
+				// * The `IgnoreIfExists` option is currently not working as expected, see https://issues.couchbase.com/browse/NCBC-2647.
```
Comment (simon-dew) [inline modules/hello-world/examples/IndexHelloWorld.csx]: Is there any chance you could put the note outside of the tagged `primary` area?

File: modules/hello-world/examples/IndexHelloWorld.csx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+				new[] { "name", "id", "icao", "iata" }
+			);
+			// end::composite[]
+			Console.WriteLine("Index creation complete");
+		}
+
+		{
+			Console.WriteLine("\n[drop-primary]");
+			// tag::drop-primary[]
+			await cluster.QueryIndexes.DropPrimaryIndexAsync("`travel-sample`");
```
Comment (simon-dew) [inline modules/hello-world/examples/IndexHelloWorld.csx:81]: Might be handy (but not necessary) to have examples for creating and dropping _named_ primary indexes (which I think is possible with the .NET SDK?)

**Review by simon-dew — COMMENTED** [review]
This is great, thanks @maria-robobug

File: modules/hello-world/examples/IndexHelloWorld.csx (hunk 2 above)
Comment (maria-robobug) [reply modules/hello-world/examples/IndexHelloWorld.csx:81]: Ah, maybe I missed that, I'll have a look! 👍

Comment (maria-robobug) [reply modules/hello-world/examples/IndexHelloWorld.csx:81]: Hmm, @simon-dew is there actually a difference between a `secondary` and a `named primary` index?
If they're the same, then we already have an example for creating this on line 44.

Comment (simon-dew) [reply modules/hello-world/examples/IndexHelloWorld.csx:81]: If my understanding of the SDK reference docs is correct, the syntax is different for named primary indexes and secondary indexes.

So with [CreateIndexAsync][1] and [DropIndexAsync][2], the `indexName` is one of the required parameters.

However with [CreatePrimaryIndexAsync][3] and [DropPrimaryIndexAsync][4], if you want to specify the index name, you have to do it using the options object.

[1]: https://docs.couchbase.com/sdk-api/couchbase-net-client/api/Couchbase.Management.Query.IQueryIndexManager.html#Couchbase_Management_Query_IQueryIndexManager_CreateIndexAsync_System_String_System_String_IEnumerable_System_String__Couchbase_Management_Query_CreateQueryIndexOptions_ 
[2]: https://docs.couchbase.com/sdk-api/couchbase-net-client/api/Couchbase.Management.Query.IQueryIndexManager.html#Couchbase_Management_Query_IQueryIndexManager_DropIndexAsync_System_String_System_String_Couchbase_Management_Query_DropQueryIndexOptions_
[3]: https://docs.couchbase.com/sdk-api/couchbase-net-client/api/Couchbase.Management.Query.IQueryIndexManager.html#Couchbase_Management_Query_IQueryIndexManager_CreatePrimaryIndexAsync_System_String_Couchbase_Management_Query_CreatePrimaryQueryIndexOptions_
[4]: https://docs.couchbase.com/sdk-api/couchbase-net-client/api/Couchbase.Management.Query.IQueryIndexManager.html#Couchbase_Management_Query_IQueryIndexManager_DropPrimaryIndexAsync_System_String_Couchbase_Management_Query_DropPrimaryQueryIndexOptions_

Comment (simon-dew) [reply modules/hello-world/examples/IndexHelloWorld.csx:81]: `CreatePrimaryIndexAsync` and `DropPrimaryIndexAsync` don't provide a _parameter_ for specifying the index name.

Comment (maria-robobug) [reply modules/hello-world/examples/IndexHelloWorld.csx:81]: Ah, that makes sense, I got both of these muddled up. So you can create a primary index type with a custom name 👍 .

File: modules/hello-world/examples/IndexHelloWorld.csx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+			// end::primary[]
+			Console.WriteLine("Index creation complete");
+		}
+
+		{
+			Console.WriteLine("\n[named-primary]");
+			// tag::named-primary[]
+			await cluster.QueryIndexes.CreatePrimaryIndexAsync(
+				"`travel-sample`",
+				options => options.IndexName("named_primary_index")
```
Comment (maria-robobug) [inline modules/hello-world/examples/IndexHelloWorld.csx:48]: I've added this, but not clear if this is correct as it doesn't seem to work, asked dotnet SDK team for clarification.
https://couchbase.slack.com/archives/CC6GX1CJJ/p1642010749048100

**Comment by simon-dew** [comment]
Just to say @maria-robobug — I have used all the tagged examples in `IndexHelloWorld.csx`, including the new `named-primary` and `drop-named-primary` tags. Thanks!

File: modules/hello-world/examples/IndexHelloWorld.csx (hunk 3 above)
Comment (maria-robobug) [reply modules/hello-world/examples/IndexHelloWorld.csx:48]: Ok, at least confirmed this is the right syntax so good to go.

**Pull request merged**